R/get.R

```diff
@@ -0,0 +1,228 @@
+  } else if (isTRUE(article == "annex")) {
+    t <- lapply(t, function(x) grep("^annex", x, ignore.case = TRUE,
+                                    value = TRUE))
+  }
+  if (!is.null(match)) {
+    t <- lapply(t, function(x) grep(match, x, ignore.case = TRUE, value = TRUE))
+  }
+  t <- ifelse(lengths(t) == 0, NA_character_, t)
+  t
+}
+
+#' Get links from manyID
+#'
+#' Treaties that modify, amend, or expand other treaties
+#' usually specify so in the title.
+#' We use the manyIDs generated by `code_agreements()` and
+#' `condense_agreements()` to return a data frame of treaty
+#' links for a database.
+#' @param database A Package database
+#' @param dataset A Package dataset
+#' @param treaty_type The type of treaties to be returned.
+#' By default, all treaties are returned.
+#' Other options include bilateral or multilateral treaties.
+#' @return A dataframe of agreements' treatyID and their linkages.
+#' @importFrom purrr map map_chr
+#' @examples
+#' \dontrun{
+#' get_links(database = manyenviron::agreements)
+#' get_links(database = manyenviron::agreements, treaty_type = "multilateral")
+#' get_links(dataset = manyenviron::agreements$IEADB, treaty_type = "bilateral")
+#' samples <- lapply(manyenviron::agreements,
+#' function(x) x[x$Beg > "1991-12-31" & x$Beg < "1993-01-01", ])
+#' migraph::gglineage(get_links(samples))
+#' }
+#' @export
+get_links <- function(database, dataset, treaty_type = "all") {
+  # Get manyID
+  if (!missing(database)) {
+    treatyID <- unname(unlist(purrr::map(database, "manyID")))
+  }
+  if (!missing(dataset)) {
+    treatyID <- dataset$manyID
+  }
+  usethis::ui_done("treatyID column in dataset automatically found")
+  # Filter by links
+  treatyID <- grep(":", treatyID, value = TRUE)
+  # Filter by treaty_type
+  if (treaty_type == "bilateral") {
+    treatyID <- grep("-", treatyID, value = TRUE)
+  }
+  if (treaty_type == "multilateral") {
+    treatyID <- grep("-", treatyID, value = TRUE, invert = TRUE)
+  }
+  # Split treatyID
+  link <- purrr::map_chr(strsplit(treatyID, ":"), 2)
+  agreement <- purrr::map_chr(strsplit(treatyID, ":"), 1)
+  # Return dataset
+  out <- data.frame(agreement, link)
+  out
+}
+
+#' Get memberships' list
+#'
+#' Memberships database have actor column(s) and
+#' treaty column(s) but information on the other
+#' countries that are party to the treaties is often
+#' hard to read.
+#' The function allows to generate a dataframe of
+#' treaty IDs and actors part to the treaty.
+#' @param database A database
+#' @param actor A actor variable (e.g. country)
+#' @param id A treaty ID variable
+#' @return A dataframe of treaty IDs and actors
+#' part of the treaty
+#' @import dplyr
+#' @examples
+#' \dontrun{
+#' sample <- manyenviron::memberships$IEADB_MEM
+#' get_memberships(actor = sample$CountryID, id = sample$manyID)
+#' get_memberships(manyenviron::memberships)
+#' }
+#' @export
+get_memberships <- function(database, actor, id) {
```

Comment:
code_memberships?

Reply:
I called `retrieved_memeberships()` for now since it does really code data but only wrangles data to a more readable/workable format if we are interested in memberships to treaties. Please let me know if this works, otherwise I can change the name again or we can defunct this function (though I think we use it for plotting agreements in a shinny app for manyenviron).